modules/local/filter_barcode.nf

```diff
@@ -2,29 +2,29 @@ process FILTER_BARCODE {
     tag "${meta.id}"
     label 'process_low'
 
-    conda "conda-forge::python=3.9"
+    conda "conda-forge::python=3.9 conda-forge::biopython=1.78"
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
-        'https://depot.galaxyproject.org/singularity/python:3.9' :
-        'biocontainers/python:3.9' }"
+        'https://depot.galaxyproject.org/singularity/biopython:1.78' :
+        'biocontainers/biopython:1.78' }"
 
     input:
-    tuple val(meta), path(fasta)
-    tuple val(meta2), path(barcodes)
+    tuple val(meta) , path(fasta)
+    tuple val(meta2), path(blast_data)
+    val barcodes
 
     output:
-    tuple val(meta), path( "*txt" ) , emit: debarcoded
-    path "versions.yml"             , emit: versions
+    tuple val(meta), path( "*filtered.txt" )    , emit: debarcoded
+    path "versions.yml"                         , emit: versions
 
     script:
-    def prefix  = task.ext.prefix   ?: "${meta.id}"
     def args    = task.ext.args     ?: ''
-    def prefix  = task.ext.prefix   ?: "${meta.id}.debarcoded" // args is for `pacbio_multiplexing_barcodes_check_${meta.barcode}.txt`
+    def prefix  = task.ext.prefix   ?: "${meta.id}"
     """
     filter_barcode_blast_results.py \\
         --input ${fasta} \\
-        --barcodes ${barcodes} \\
+        --barcode ${barcodes} \\
         --blast ${blast_data} \\
-        --output ${prefix}.txt
+        --output ${prefix}-${barcodes}-filtered.txt
```

Comment:
${prefix}-${barcodes}-filtered.txt change to ${prefix}_${barcodes}_filtered.txt

Reply:
Done